Caboodle/Permissions/Permissions.android.cs

```diff
@@ -0,0 +1,151 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Android;
+using Android.Content.PM;
+using Android.Support.V4.App;
+using Android.Support.V4.Content;
+
+namespace Microsoft.Caboodle
+{
+    internal static partial class Permissions
+    {
+        static readonly object locker = new object();
+
+        static Dictionary<PermissionType, (int requestCode, TaskCompletionSource<PermissionStatus> tcs)> requests =
+            new Dictionary<PermissionType, (int, TaskCompletionSource<PermissionStatus>)>();
```

Comment:
Maybe make this a `using PermissionsDictionary = Sys.Coll.Gen.Dictionary<PermissionType, (int requestCode, TaskCompletionSource<PermissionStatus> tcs)>;` to make it easier to read as this is a very long type.
Maybe?